rules/Php80/Rector/ClassMethod/AddParamBasedOnParentClassMethodRector.php

```diff
@@ -104,12 +104,12 @@ public function refactor(Node $node): ?Node
             return null;
         }
 
-        $parentClassMethod = $this->astResolver->resolveClassMethodFromMethodReflection($parentMethodReflection);
```

Comment:
`astResolver->resolveClassMethodFromMethodReflection` is slow as can be seen in https://github.com/rectorphp/rector-src/pull/4804

we can do the `isPrivate` check earlier based on `MethodReflection` and therefore reduce the number of times we need to invoke `astResolver->resolveClassMethodFromMethodReflection`